challenge-manager/edge-tracker/tracker.go

```diff
@@ -379,94 +453,108 @@ func (et *Tracker) uniqueTrackerLogFields() []any {
 }
 
 func (et *Tracker) tryToConfirmEdge(ctx context.Context) (bool, error) {
-	// If the edge is not a root, block challenge edge, we have nothing to do here.
-	if !IsRootBlockChallengeEdge(et.edge) {
+	fields := et.uniqueTrackerLogFields()
+	// If the edge is not a root of a challenge or subchallenge, we have nothing to do here.
+	if et.edge.ClaimId().IsNone() {
+		return false, nil
+	}
+	if strings.Contains(et.validatorName, "evil") {
 		return false, nil
 	}
 	status, err := et.edge.Status(ctx)
 	if err != nil {
+		log.Error("Could not get edge status", fields...)
 		return false, errors.Wrap(err, "could not get edge status")
 	}
 	if status == protocol.EdgeConfirmed {
+		log.Error("Could already confirmed", fields...)
 		return true, nil
 	}
-	assertionHash, err := et.edge.AssertionHash(ctx)
+	challengedAssertionHash, err := et.edge.AssertionHash(ctx)
 	if err != nil {
+		log.Error("Could not get challenged assertion hash", fields...)
 		return false, err
 	}
-	fields := et.uniqueTrackerLogFields()
-	start := time.Now()
-	computedTimer, err := et.chainWatcher.ComputeRootInheritedTimer(ctx, assertionHash)
-	if err != nil {
-		log.Error("Could not update time cache", "err", err)
-		return false, errors.Wrap(err, "could not update edge inherited timer")
-	}
-	end := time.Since(start)
-	onchainTimer, err := et.edge.SafeHeadInheritedTimer(ctx)
-	if err != nil {
-		return false, errors.Wrap(err, "could not get edge onchain inherited timer")
-	}
 	manager, err := et.chain.SpecChallengeManager(ctx)
 	if err != nil {
+		log.Error("Could not get challenge manager", fields...)
 		return false, errors.Wrap(err, "could not get challenge manager")
 	}
 	chalPeriod, err := manager.ChallengePeriodBlocks(ctx)
 	if err != nil {
+		log.Error("Could not get challenge period blocks", fields...)
 		return false, errors.Wrap(err, "could not check the challenge period length")
 	}
+	start := time.Now()
+	isConfirmable, _, computedTimer, err := et.chainWatcher.IsConfirmableEssentialNode(
+		ctx,
+		challengedAssertionHash,
+		et.edge.Id(),
+		chalPeriod,
+	)
+	if err != nil {
+		log.Error("Could not check if essential node is confirmable")
```

Comment:
If you agree with earlier review comments, can we also say "essential edge" here and on the next line?